### PR TITLE
Fix macOS requirement in Sierra Cache Cleaner.app Cask

### PR DESCRIPTION
**Pull request by Saklad5**
### Checklist
- [x] The commit message includes the cask’s name and version.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.

This application is supported on OS X 10.4 and later, but is hard-coded to not work on anything newer than macOS Sierra.

### CODE REVIEW

File: Casks/sierra-cache-cleaner.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@
   homepage 'http://www.northernsoftworks.com/sierracachecleaner.html'
   license :commercial
 
-  depends_on macos: :sierra
+  depends_on macos: '<= :sierra'
```
Comment (vitorgalvao) [inline Casks/sierra-cache-cleaner.rb]: This makes no sense. If you want to do this, simply delete the line.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/sierra-cache-cleaner.rb (hunk 1 above)
Comment (reitermarkus) [inline Casks/sierra-cache-cleaner.rb]: The thing is that it will stop working on `> :sierra`.

Comment (reitermarkus) [inline Casks/sierra-cache-cleaner.rb]: Then again, it is a `:latest` cask, so it will likely get updated in time.

**Comment by vitorgalvao** [comment]
> This application is supported on OS X 10.4 and later, but is hard-coded to not work on anything newer than macOS Sierra.

Yes, because [you pointed out it didn’t work in other versions](https://github.com/caskroom/homebrew-cask/pull/24748#discussion_r80224245). So it does work in older versions? Even with `Sierra` specifically in the name?

**Comment by Saklad5** [comment]
It is incredibly confusing. This is the support page detailing this: http://www.northernsoftworks.com/support.html

As you can see, each version of the app stops working when a new major release of macOS comes out.

**Comment by reitermarkus** [comment]
This has to be the worst naming convention I have seen so far:

> Download Sierra Cache Cleaner 11.0.1 (72.9 Mb) macOS X 10.4-10.12.1
> Download El Capitan Cache Cleaner 10.0.6 (61.2 Mb) OS X 10.4 - 10.11.6
> Download Yosemite Cache Cleaner 9.0.8 (58.2 Mb) OS X 10.4-10.10.5
> Download Mavericks Cache Cleaner 8.0.9 (31.1 Mb) OS X 10.4-10.9.5
> Download Mountain Lion Cache Cleaner 7.0.10 (20.4 Mb) OS X 10.4-10.8
> Download Lion Cache Cleaner 6.0.13 (19.3 Mb) OS X 10.4-10.7
> Download Snow Leopard Cache Cleaner 5.0.22 (18.7 Mb) OS X 10.3.9-10.6
> Download Leopard Cache Cleaner 4.0.27 (14 Mb) OS X 10.3.9 - 10.5
> Download Tiger Cache Cleaner 3.2.19 (9.4 Mb) OS X 10.3.9 - 10.4
> Download Panther Cache Cleaner 2.4.8 (3.6 Mb) OS X 10.1 - 10.3
> Download TCC 3.0.7 for OS X 10.1

**Comment by Saklad5** [comment]
It does a lot of stuff that is practically guaranteed to screw up with each major OS update, so the developers basically rework the whole thing and sell it separately.

As it could fail catastrophically on macOS versions it doesn't support, it is pre-emptively hard-coded to not work on the next major release.

**Comment by vitorgalvao** [comment]
This one will be bothersome to update. Not looking forward to having a bunch of similarly named casks filling the repo, or renaming this one every time. Not sure we should keep this one.

**Comment by Saklad5** [comment]
Really? It shouldn't be difficult to maintain. It's not like this cask will break when macOS is updated again, and we could always move older ones to /versions if clutter becomes an issue.




            Yours truly,
                              Jeremy Saklad

On Sep 30, 2016, at 3:51 PM, Vítor Galvão <notifications@github.com> wrote:

This one will be bothersome to update. Not looking forward to having a bunch of similarly named casks filling the repo, or renaming this one every time. Not sure we should keep this one.

**Comment by vitorgalvao** [comment]
I didn’t say it would be difficult, I said it would be a bother.

Yes, of course, we have the versions repo, but it feels like we’d be treating it as a bit of a dumping ground. However, I just remembered [we have rules to prevent that](https://github.com/caskroom/homebrew-versions#acceptable-casks), so it’s fine.

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**